src/ripple/overlay/impl/PeerSet.cpp

```diff
@@ -73,8 +73,15 @@ bool PeerSet::insert (std::shared_ptr<Peer> const& ptr)
 void PeerSet::setTimer ()
 {
     mTimer.expires_from_now(mTimerInterval);
-    mTimer.async_wait (std::bind (&PeerSet::timerEntry,
-        pmDowncast (), beast::asio::placeholders::error));
+    mTimer.async_wait (
+        [wptr=pmDowncast()](boost::system::error_code const& ec)
+        {
+            if (ec == boost::asio::error::operation_aborted)
+                return;
+
+            if (auto ptr = wptr.lock ())
+                ptr->execute ();
+        });
```

Comment:
While I was reviewing a previous commit I did a double take with binding a `weak_ptr` to what looked like a member function. The lambda makes this much clearer (and thanks for the exclamation point in the commit message :-)